src/ModalCategory.js

```diff
@@ -1,49 +1,45 @@
-import React, {Component} from 'react';
+import React from 'react';
 import './Modal.css';
 
-class ModalCategory extends Component {
-
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            modalInput: this.props.text
-        };
-
-        this.handleInputChange = this.handleInputChange.bind(this);
-        this.handleSubmit = this.handleSubmit.bind(this);
-    }
-
-    handleInputChange (e) {
-        this.setState({ modalInput: e.target.value });
-    }
-
-
-    handleSubmit () {
-        this.props.actionSubmit(this.state.modalInput);
-    }
+const ModalCategory = ({ id, name, isActive, type, onSubmitEditCategory, onSubmitAddSubcategory, onHideModal }) => {
+
+    let inputName;
+
+    const handleSubmit = () => {
+        if(type === 'add') {
+            onSubmitAddSubcategory(inputName.value, id);
+        } else if(type === 'edit') {
+            onSubmitEditCategory(id, inputName.value);
+        }
+    };
```

Comment:
Avoid creating functions inside render - it will be created on each (re)render.

You can simplify this. For example:
- leave only one `onSubmit` handler here, that will accept three arguments (type, name, id)
- decide if it's adding or editing inside action (basing on type value)